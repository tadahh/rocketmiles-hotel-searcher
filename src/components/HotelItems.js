import React from 'react';
import {
    Card,
    Header,
    Image,
    Grid,
    Segment,
    Icon,
    Label,
    Statistic
} from 'semantic-ui-react';

const HotelItems = ({ hotelData }) => {
    const decodeHtml = html => {
        var txt = document.createElement('textarea');
        txt.innerHTML = html;
        return txt.value;
    };

    return (
        <Segment basic>
            <Card.Group>
                {hotelData.map(hotel => {
                    return (
                        <Card fluid key={hotel.id}>
                            <Grid stretched>
                                <Grid.Column width={5}>
                                    <Segment basic>
                                        <Image
                                            src={
                                                hotel.hotelStaticContent
                                                    .mainImage.url
                                            }
                                            style={{
                                                height: 210,
                                                width: '100%'
                                            }}
                                            rounded
                                        />
                                    </Segment>
                                </Grid.Column>

                                <Grid.Column width={7}>
                                    <Segment basic>
                                        <Header size='medium'>
                                            <a>
                                                <Icon
                                                    color='blue'
                                                    name='map marker alternate'
                                                />
                                                {
                                                    hotel.hotelStaticContent
                                                        .neighborhoodName
                                                }
                                            </a>
                                        </Header>

                                        <Header size='large'>
                                            {hotel.hotelStaticContent.name}
                                        </Header>
                                    </Segment>
                                    <Segment basic>
                                        <Statistic.Group size='mini'>
                                            <Statistic>
                                                <Statistic.Value>
                                                    {
                                                        hotel.hotelStaticContent
                                                            .stars
                                                    }
                                                </Statistic.Value>
                                                <Statistic.Label>
                                                    Stars
                                                </Statistic.Label>
                                            </Statistic>
                                            <Statistic>
                                                <Statistic.Value>
                                                    {
                                                        hotel.hotelStaticContent
                                                            .numberOfReviews
                                                    }
                                                </Statistic.Value>
                                                <Statistic.Label>
                                                    Reviews
                                                </Statistic.Label>
                                            </Statistic>
                                        </Statistic.Group>
                                    </Segment>
                                </Grid.Column>
                                <Grid.Column width={4} textAlign='right'>
                                    <Segment basic>
                                        <Label color='blue'>
                                            {`Earn ${
                                                hotel.rewards.miles
                                            } Miles`}
                                        </Label>
                                        <Header size='large' color='green'>
                                            {`${decodeHtml(
                                                hotel.lowestAveragePrice.symbol
                                            )}${
                                                hotel.lowestAveragePrice.amount
                                            }`}
                                        </Header>
                                    </Segment>
                                </Grid.Column>
                            </Grid>
                        </Card>
                    );
                })}
            </Card.Group>
        </Segment>
    );
};

export default HotelItems;
