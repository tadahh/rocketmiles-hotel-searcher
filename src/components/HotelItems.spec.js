import React from 'react';
import { shallow, mount } from 'enzyme';
import HotelItems from './HotelItems';

const hotelData = [
    {
        id: '907',
        rewards: {
            miles: 10000
        },
        lowestAveragePrice: {
            currency: 'USD',
            symbol: '&#36;',
            amount: 579
        },
        hotelStaticContent: {
            hotelId: 907,
            languageCode: 'en',
            mainImage: {
                category: 'EXTERIOR',
                url:
                    'http://d2whcypojkzby.cloudfront.net/imageRepo/2/0/68/56/314/ExteriorCarsGone_S.jpg',
                source: 'VFML'
            },
            name: 'Omni Chicago Hotel & Suites Magnificent Mile',
            neighborhoodName: 'Magnificent Mile',
            address: {
                line1: '676 North Michigan Avenue',
                line2: null,
                city: 'Chicago',
                stateName: 'Illinois',
                stateCode: 'IL',
                countryName: 'United States',
                countryCode: 'US',
                zipCode: '60611',
                latitude: 41.89475,
                longitude: -87.62465,
                timeZoneId: 'America/Chicago'
            },
            stars: 4,
            rating: 9,
            numberOfReviews: 685,
            brandCode: '1324',
            brandName: 'Omni Hotels',
            propertyType: 204,
            propertyTypeName: 'Hotel'
        }
    }
];

describe('HotelItems', () => {
    const wrapper = mount(<HotelItems hotelData={hotelData} />);

    it('accepts hotelData props', () => {
        expect(wrapper.props().hotelData).toEqual(hotelData);
    });

    it('renders', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('returns default empty array when there is no data', () => {
        const wrapper = mount(<HotelItems hotelData={[]} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('does not break without data', () => {
        const wrapper = mount(<HotelItems hotelData={[]} />);
        expect(wrapper.find('.card')).toHaveLength(0);
    });

    it('contains image src for hotel photo', () => {
        expect(wrapper.find('img').prop('src')).toEqual(
            hotelData[0].hotelStaticContent.mainImage.url
        );
    });

    it('contains neighborhood name', () => {
        expect(
            mount(
                <div className='ui medium header'>
                    <a>
                        <i
                            aria-hidden='true'
                            className='blue map marker alternate icon'
                        />
                        Magnificent Mile
                    </a>
                </div>
            ).text()
        ).toEqual(hotelData[0].hotelStaticContent.neighborhoodName);
    });
});
