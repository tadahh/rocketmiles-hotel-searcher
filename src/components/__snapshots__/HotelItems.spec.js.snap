// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelItems renders 1`] = `ReactWrapper {}`;

exports[`HotelItems returns default empty array when there is no data 1`] = `ReactWrapper {}`;
